### PR TITLE
fix: use browseResults for results-per-page

### PR DESCRIPTION
**Pull request by louis-bompart**
https://coveord.atlassian.net/browse/KIT-3387

### CODE REVIEW

File: packages/headless/src/controllers/results-per-page/headless-results-per-page.ts [hunk 1]
```diff
@@ -52,7 +55,7 @@ export function buildResultsPerPage(
 
     set(num: number) {
       coreController.set(num);
-      dispatch(fetchPage({legacy: logPagerResize()}));
+      dispatch(fetchPage({legacy: logPagerResize(), next: browseResults()}));
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/results-per-page/headless-results-per-page.ts:58]: I think we should adjust the tests accordingly

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/headless/src/controllers/results-per-page/headless-results-per-page.ts (hunk 1 above)
Comment (louis-bompart) [reply packages/headless/src/controllers/results-per-page/headless-results-per-page.ts:58]: All tests are green tho 🙃 ?

Comment (fbeaudoincoveo) [reply packages/headless/src/controllers/results-per-page/headless-results-per-page.ts:58]: Yes, but the fact that you can change the `fetchPage` action payload and the tests are still green seem to indicate that the test is incomplete

Comment (louis-bompart) [reply packages/headless/src/controllers/results-per-page/headless-results-per-page.ts:58]: Fair, I added a basic ut to cover the change in parameters at the ut level.
More advanced/comprehensive testing might be wise, but to be done properly should probably be done either from an e2e (true e2e, that includes the backend systems) perspective, or thru contract testing.
The former would be sluggish and not really setup, the second one would probably be the most adequate, but contract testing is not something done at Coveo yet and span across multiple teams, so out-of-scope.

**Pull request merged**